tests/foreman/api/test_subnet.py

```diff
@@ -21,528 +21,528 @@
 from requests.exceptions import HTTPError
 
 from robottelo.datafactory import gen_string
 from robottelo.datafactory import generate_strings_list
 from robottelo.datafactory import invalid_values_list
-from robottelo.test import APITestCase
+from robottelo.datafactory import parametrized
 
 
-class ParameterizedSubnetTestCase(APITestCase):
-    """Implements parametrized subnet tests in API"""
+@pytest.mark.tier1
+def test_positive_create_with_parameter():
+    """Subnet can be created along with parameters
 
-    @pytest.mark.tier1
-    def test_positive_create_with_parameter(self):
-        """Subnet can be created along with parameters
+    :id: ec581cb5-8c48-4b9c-b536-302c0b7ec30f
 
-        :id: ec581cb5-8c48-4b9c-b536-302c0b7ec30f
+    :steps: Create Subnet with parameter that has single key and single
+        value
 
-        :steps: Create Subnet with parameter that has single key and single
-            value
+    :expectedresults: The Subnet is created with parameter
+    """
+    parameter = [{'name': gen_string('alpha'), 'value': gen_string('alpha')}]
+    subnet = entities.Subnet(subnet_parameters_attributes=parameter).create()
+    assert subnet.subnet_parameters_attributes[0]['name'] == parameter[0]['name']
+    assert subnet.subnet_parameters_attributes[0]['value'] == parameter[0]['value']
+
+
+@pytest.mark.tier1
+@pytest.mark.parametrize('name', **parametrized(generate_strings_list()))
+def test_positive_add_parameter(name):
+    """Parameters can be created in subnet
+
+    :id: c1dae6f4-45b1-45db-8529-d7918e41a99b
+
+    :parametrized: yes
+
+    :steps:
+
+        1. Create Subnet with all the details
+        2. Create subnet parameter with single key and single value
+
+    :expectedresults: The parameter should be created in subnet
+
+    :CaseImportance: Medium
+    """
+    subnet = entities.Subnet().create()
+    value = gen_string('utf8')
+    subnet_param = entities.Parameter(subnet=subnet.id, name=name, value=value).create()
+    assert subnet_param.name == name
+    assert subnet_param.value == value
+
+
+@pytest.mark.tier1
+def test_positive_add_parameter_with_values_and_separator():
+    """Subnet parameters can be created with values separated by comma
+
+    :id: b3de6f96-7c39-4c44-b91c-a6d141f5dd6a
+
+    :steps:
+
+        1. Create Subnet with all the details
+        2. Create subnet parameter having single key and values
+            separated with comma
+
+    :expectedresults: The parameter with values separated by comma should
+        be saved in subnet
+
+    :CaseImportance: Low
+    """
+    subnet = entities.Subnet().create()
+    name = gen_string('alpha')
+    values = ', '.join(generate_strings_list())
+    subnet_param = entities.Parameter(name=name, subnet=subnet.id, value=values).create()
+    assert subnet_param.name == name
+    assert subnet_param.value == values
+
+
+@pytest.mark.tier1
+@pytest.mark.parametrize('separator', **parametrized([',', '/', '-', '|']))
```

Comment:
here we go :) @JacobCallahan

Reply:
This works, thanks!

When you pass a dictionary to `parametrized` method, it will use the dict keys as parameter ids - not necessary here, but optional.

@pytest.mark.parametrize('separator', **parametrized({'comma': ',', 'slash': '/', 'dash': '-', 'pipe': '|'}))

Reply:
I mean does it really matter ? Sure it can save you some time, once you will see the test failed with parameter dash. 
But I bet this test will not fail. 

I mark the change below :)